### PR TITLE
Inline ScalarFormatter._formatSciNotation.

### PR DESCRIPTION
**Pull request by anntzer**
It's only used in one place, in a tiny function, and doesn't have a
particularly intuitive name, so it's actually easier to follow the logic
by having it all in one place.  Moreover the implementation can be
greatly simplified by separately computing and formatting the mantissa
and the exponent, rather than trying to parse the locale-dependent
result of the formatting of the entire number.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by dopplershift — APPROVED** [review]

**Comment by jklymak** [comment]
The doc error looks real to me...

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -730,19 +733,23 @@ def format_data_short(self, value):
...
-            self.fix_minus(
-                locale.format_string(fmt, (value,)) if self._useLocale else
-                fmt % value))
+        return self._format_maybe_minus_and_locale(fmt, value)
 
     def format_data(self, value):
         # docstring inherited
-        if self._useLocale:
-            s = locale.format_string('%1.10e', (value,))
+        e = math.floor(math.log10(value))
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py:740]: Probably needs to be `abs(value)` to fix the docs.

File: lib/matplotlib/ticker.py [hunk 2]
```diff
@@ -730,19 +733,23 @@ def format_data_short(self, value):
...
+        s = round(value / 10**e, 10)
+        exponent = self._format_maybe_minus_and_locale("%d", e)
+        significand = self._format_maybe_minus_and_locale(
+            "%d" if s % 1 == 0 else "%1.10e", s)
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py:744]: Is `%e` really the right format now that `s` is divided by the exponent?

File: lib/matplotlib/ticker.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/ticker.py:740]: thanks

File: lib/matplotlib/ticker.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/ticker.py]: I don't think it matters?  but I agree %f may be clearer.

**Review by QuLogic — APPROVED** [review]

**Comment by vpicouet** [comment]
Hello,
I used to use `_formatSciNotation` to convert number to scienctific notation (as mentionned here https://stackoverflow.com/questions/25750170/show-decimal-places-and-scientific-notation-on-the-axis-of-a-matplotlib-plot ):
`fmt = mticker.FuncFormatter(lambda x, pos: "${}$".format(mticker.ScalarFormatter(useOffset=False, useMathText=True)._formatSciNotation("%1.10e" % np.round(x, 5))))
`
What is the best way to do it now?
Thanks for your help!

**Comment by jklymak** [comment]
@vpicouet please ask usage questions at discourse.matplotlib.org.  However this is as a private method so it was not meant to be used by downstream users.  You could look at the source and copy it.

**Pull request merged**